Apps/W1/Shopify/app/src/Payments/Codeunits/ShpfyPayments.Codeunit.al

```diff
@@ -203,4 +203,47 @@ codeunit 30169 "Shpfy Payments"
             end;
         end;
     end;
+
+    internal procedure UpdateDisputeStatus()
+    var
+        PaymentTransaction: Record "Shpfy Payment Transaction";
+        DisputeToken: JsonToken;
+        JResponse: JsonToken;
+        Url: Text;
+        UrlTxt: Label 'shopify_payments/disputes/%1.json', Locked = true;
+    begin
+        PaymentTransaction.SetRange(Type, PaymentTransaction.Type::Dispute);
+        PaymentTransaction.SetFilter("Dispute Status", '<>%1&<>%2', PaymentTransaction."Dispute Status"::Won, PaymentTransaction."Dispute Status"::Lost);
+
+        if PaymentTransaction.FindSet() then
+            repeat
+                Url := CommunicationMgt.CreateWebRequestURL(StrSubstNo(UrlTxt, PaymentTransaction.Id));
+                JResponse := CommunicationMgt.ExecuteWebRequest(Url, 'GET', JResponse);
+                DisputeToken := JsonHelper.GetJsonToken(JResponse, 'dispute');
+                UpdateDisputeStatus(PaymentTransaction, DisputeToken);
+            until PaymentTransaction.Next() = 0;
+    end;
+
+    internal procedure UpdateDisputeStatus(PaymentTransaction: Record "Shpfy Payment Transaction"; DisputeToken: JsonToken)
```

Comment:
This won't work without **_var_** PaymentTransaction: Record "Shpfy Payment Transaction", the payment transaction record will never get updated

Reply:
I beg to differ. The modify is done within the procedure, at the same place where the changes are done. The outside function wouldn't be aware if something was changed, however all it is doing is feeding the data into the procedure, so it doesn't really care.